src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import core.basesyntax.figure.Figure;
+import core.basesyntax.tools.FigureSupplier;
+import java.util.Random;
+
+public class Main {
+    public static void main(String[] args) {
+
+        FigureSupplier figureCreater = new FigureSupplier();

```

Comment:
This should be class level variable (https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#think-about-which-variables-should-be-local-in-the-method-and-which-should-be-class-level)